README.md

```diff
@@ -1,367 +1,654 @@
+;
+
+// The status report is now dispatched (it is still returned if you need it).
 ```
 
-### I must use PHP 5.4 or 5.5. What can I do?
-You won't be able to send any payload, so you'll only be able to use `sendOneNotification($subscription)` or `queueNotification($subscription)`.
-Install the library with `composer` using `--ignore-platform-reqs`.
-The workaround for getting the payload is to fetch it in the service worker ([example](https://github.com/Minishlink/physbook/blob/2ed8b9a8a217446c9747e9191a50d6312651125d/web/service-worker.js#L75)). 
+## Complete Example
 
-### I lost my VAPID keys!
-See [issue #58](https://github.com/web-push-libs/web-push-php/issues/58).
 
-### I'm using Firebase push notifications, how do I use this library?
-This library is not designed for Firebase push notifications.
-You can still use it for your web projects (for standard WebPush notifications), but you should forget any link to Firebase while using the library.
+```php
+<?php
 
-### I need to send notifications to native apps. (eg. APNS for iOS)
-WebPush is for web apps.
-You need something like [RMSPushNotificationsBundle](https://github.com/richsage/RMSPushNotificationsBundle) (Symfony).
+declare(strict_types=1);
+
+use Jose\Component\Core\JWK;
+use Minishlink\WebPush\Action;
+use Minishlink\WebPush\ExtensionManager;
+use Minishlink\WebPush\Message;
+use Minishlink\WebPush\Notification;
+use Minishlink\WebPush\Payload\AES128GCM;
+use Minishlink\WebPush\Payload\AESGCM;
+use Minishlink\WebPush\Payload\PayloadExtension;
+use Minishlink\WebPush\PreferAsyncExtension;
+use Minishlink\WebPush\Subscription;
+use Minishlink\WebPush\TopicExtension;
+use Minishlink\WebPush\TTLExtension;
+use Minishlink\WebPush\UrgencyExtension;
+use Minishlink\WebPush\VAPID\LcobucciProvider;
+use Minishlink\WebPush\VAPID\VAPIDExtension;
+use Minishlink\WebPush\VAPID\WebTokenProvider;
+use Minishlink\WebPush\WebPush;
 
-### This is PHP... I need Javascript!
-This library was inspired by the Node.js [marco-c/web-push](https://github.com/marco-c/web-push) library.
+use Psr\Log\LoggerInterface;
+use Psr\EventDispatcher\EventDispatcherInterface;
+use Psr\Http\Client\ClientInterface;
+use Psr\Http\Message\RequestFactoryInterface;
+use Psr\Cache\CacheItemPoolInterface;
+
+//PSR-3 Logger
+/** @var LoggerInterface $logger */
+$logger = '…';
+
+//PSR-6 Cache
+/** @var CacheItemPoolInterface $cache */
+$logger = '…';
+
+//PSR-14 Event Dispatcher
+/** @var EventDispatcherInterface $eventDispatcher */
+$eventDispatcher = '…';
+
+//PSR-17 request factory
+/** @var RequestFactoryInterface $requestFactory */
+$requestFactory = '…';
+
+//PSR-18 HTTP client
+/** @var ClientInterface $client */
+$client = '…';
+
+
+$serverPublicKey = 'BB4W1qfBi7MF_Lnrc6i2oL-glAuKF4kevy9T0k2vyKV4qvuBrN3T6o9-7-NR3mKHwzDXzD3fe7XvIqIU1iADpGQ';
+$serverPrivateKey = 'C40jLFSa5UWxstkFvdwzT3eHONE2FIJSEsVIncSCAqU';
+
+//Payload Extension
+$payloadExtension = PayloadExtension::create()
```

Comment:
It looks also more complicated than it should. The web-push library should add the necessary content encodings, it's not up to the user of the library

Reply:
Understood.
I added a `SimpleWebPush` that does all the job:

```php

$webPush = SimpleWebPush::create($client, $requestFactory)
    ->enableVapid(                         //Mandatory for some Push Services
        'http://localhost:8000',
        'BB4W1qfBi7MF_Lnrc6i2oL-glAuKF4kevy9T0k2vyKV4qvuBrN3T6o9-7-NR3mKHwzDXzD3fe7XvIqIU1iADpGQ',
        'C40jLFSa5UWxstkFvdwzT3eHONE2FIJSEsVIncSCAqU'
    )
    ->setLogger($logger)                   //Optional
    ->setCache($cache)                     //Optional
    ->setEventDispatcher($eventDispatcher) //Optional

    ->send($notification, $subscription)
;
```

Much simpler right?